src/protocols/aeternity/libs/AeAccountHdWallet.ts

```diff
@@ -89,6 +94,96 @@ export class AeAccountHdWallet extends MemoryAccount {
+    if (options?.aeppOrigin) {
+      const dataType = new TypeResolver().resolveType(aci);
+      const decodedData = new ContractByteArrayEncoder().decodeWithType(data, dataType);
+      const {
+        name, version, networkId, contractAddress,
+      } = options;
+      const opt = {
+        name, version, networkId, contractAddress, aci, data, decodedData,
+      };
+      const bigintReplacer = (k: string, v: any) => (
+        typeof v === 'bigint' ? `${v} (as BigInt)` : v
+      );
+      const message = `sign typed data:\n${JSON.stringify(opt, bigintReplacer, 2)}`;
+      const { checkOrAskPermission } = usePermissions();
+      const permissionGranted = await checkOrAskPermission(
+        METHODS.signTypedData,
+        options.aeppOrigin,
+        { message },
+      );
+      if (!permissionGranted) {
+        throw new RpcRejectedByUserError();
+      }
+    }
+
+    return super.signTypedData(
+      data,
+      aci,
+      {
+        ...options, // Mainly to pass the `fromAccount` property
+        aeppOrigin: undefined,
+      },
+    );
+  }
+
+  override async signDelegation(
+    delegation: Encoded.Bytearray,
+    options: Parameters<AccountBase['signDelegation']>[1] = {},
+  ): Promise<Encoded.Signature> {
+    let message;
+    let resolvedName;
+    const { getMiddleware } = useAeMiddleware();
+    if (options?.aeppOrigin) {
+      const params = unpackDelegation(delegation);
+      switch (params.tag) {
+        case DelegationTag.AensName:
+          try {
+            resolvedName = (await (await getMiddleware()).getName(params.nameId)).name;
+          } catch (e) {
+            handleUnknownError(e);
+          }
+          message = `sign delegation of ${resolvedName || params.nameId} to ${params.contractAddress}`;
+          break;
+        case DelegationTag.AensPreclaim:
+          message = `sign delegation of name preclaim to ${params.contractAddress}`;
+          break;
+        case DelegationTag.AensWildcard:
+          message = `sign delegation of name claiming to ${params.contractAddress}`;
+          break;
+        case DelegationTag.Oracle:
+          message = `sign delegation to allow ${params.contractAddress} operate oracle binded to current account`;
+          break;
+        case DelegationTag.OracleResponse:
+          message = `sign delegation of ${params.queryId} to ${params.contractAddress}`;
+          break;
+        default:
+          message = 'failed to distinguish delegation type';
+      }
+      const { checkOrAskPermission } = usePermissions();
+      const permissionGranted = await checkOrAskPermission(
+        METHODS.signDelegation,
+        options.aeppOrigin,
+        { message },
+      );
+      if (!permissionGranted) {
+        throw new RpcRejectedByUserError();
+      }
+    }
+
+    return super.signDelegation(delegation, {
+      ...options, // Mainly to pass the `fromAccount` property
+      networkId: this.nodeNetworkId.value,
+      aeppOrigin: undefined,
```

Comment:
```suggestion
      networkId: this.nodeNetworkId.value,
```
MemoryAccount won't pass these options to `sign` method
https://github.com/aeternity/aepp-sdk-js/blob/2743e4af828902d90e532447b6c526c833869e49/src/account/Memory.ts#L207-L217